File: python/digital_input.py
'''
Program: digital_input.py
Description: Read a digital input and display its value
Author: Aldo Nunez
'''

import mraa as m

# main
def main ():
    PORT = 5

    # initialize variable with a port number
    port_input = m.Gpio ( PORT )

    # set gpio direction to input
    port_input.dir ( m.DIR_IN )

    # read port value
    value  = port_input.read ()

    print ( "Input port %d: %d" % (  PORT , value ) )

if __name__ == "__main__":
    main ()

File: javaScript/digital_input.js
// Program: digital_input.js
// Description: Reads a digital input and displays its value
// Author: Aldo Nunez

const PORT = 5;

var m = require ( 'mraa' );

// initialize variable with a port number
var  port_input = new m.Gpio ( PORT );

// sets the gpio direction
port_input.dir ( m.DIR_IN );

// Reads the digital value
var value = port_input.read ();

console.log ( "Input port " + PORT + ": " + value );

delete ( port_input );